wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -37,9 +38,16 @@ abstract class LockedWallet
     for (utxos <- spendingInfoDAO.findAll())
       yield {
         val filtered = utxos
-          .collect {
-            case (txo) if !txo.spent && predicate(txo) =>
-              txo.output.value
+          .filter(predicate)
+          .map {
+            case txo: SpendingInfoDb =>
+              txo.state match {
+                case TxoState.UnconfirmedReceived | TxoState.ConfirmedReceived =>
+                  txo.output.value
+                case TxoState.UnconfirmedSpent | TxoState.ConfirmedSpent |
+                     TxoState.DoesNotExist =>
+                  CurrencyUnits.zero
```

Comment:
Why `sum`  zeros?

Reply:
It's a `.map` so we need to return some value. We want to ignore the values of outputs that are `Spent` or `DoesNotExist`.